### PR TITLE
Enable retrieving shadow prices for all dispatch variables

### LINKED ISSUE #180: add shadow prices for fix generators
`pypsa` does not give back shadow prices for non-extendable generators, however for non-extendable branches. One could generalize this behaviour to all components by defining the `define_dispatch_for_non_extendable_variables` function as 

```python 
def define_dispatch_for_non_extendable_variables(n, sns, c, attr):
    fix_i = get_non_extendable_i(n, c)
    if c == 'Generator':
        fix_i = fix_i.difference(n.generators.query('committable').index)
    if fix_i.empty: return
    nominal_fix = n.df(c)[nominal_attrs[c]][fix_i]
    min_pu, max_pu = get_bounds_pu(n, c, sns, fix_i, attr)
    lower = min_pu.mul(nominal_fix)
    upper = max_pu.mul(nominal_fix)
    axes = [sns, fix_i]
    dispatch = define_variables(n, -inf, inf, c, attr, axes=axes, spec='non_ext')
    dispatch = linexpr((1, dispatch))
    define_constraints(n, dispatch, '>=', lower, c, 'mu_lower', spec='non_ext')
    define_constraints(n, dispatch, '<=', upper, c, 'mu_upper', spec='non_ext')

```
in the nomopyomo setting 

This is useful for understanding the composition of the market price at a given bus.

### PR DESCRIPTION
**Pull request by FabianHofmann**
# Summary 

linopf: - define variables and constraints for fix components
 	  - remove unused arguments in lopf function
linopt: - better handling of references for static variables

Closes #180 

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- ~[ ] Unit tests for new features were added (if applicable).~
- ~[ ] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).~
- [x] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.

### CODE REVIEW

File: doc/release_notes.rst [hunk 1]
```diff
@@ -24,6 +24,7 @@ Upcoming Release
 
 * When solving ``n.lopf(pyomo=False, store_basis=True, solver_name="cplex")`` an error raised by trying to store a non-existing basis is caught.
 
+* When solving ``n.lopf(pyomo=False)``, pypsa is now constraining the dispatch variables for non extendable components with actual constraints, not with standard variable bounds. This allows for retrieving shadow prices for all dispatch variables when running ``n.lopf(pyomo=False, keep_shadowprices=True).
```
Comment (fneum) [inline doc/release_notes.rst:27]:
```suggestion
* When solving ``n.lopf(pyomo=False)``, PyPSA now constrains the dispatch variables for non extendable components with actual constraints, not with standard variable bounds. This allows retrieving shadow prices for all dispatch variables when running ``n.lopf(pyomo=False, keep_shadowprices=True).   
```

File: pypsa/linopf.py [hunk 2]
```diff
@@ -906,8 +899,7 @@ def network_lopf(n, snapshots=None, solver_name="cbc",
     #disable logging because multiple slack bus calculations, keep output clean
     snapshots = _as_snapshots(n, snapshots)
     n.calculate_dependent_values()
-    if not skip_pre:
-        n.determine_network_topology()
+    n.determine_network_topology()
```
Comment (fneum) [inline pypsa/linopf.py:902]: Any particular reason to remove this? I seem to recall a case where you wouldn't want to recalculate the sub_networks.

File: pypsa/linopf.py [hunk 3]
```diff
@@ -818,11 +815,11 @@ def v_ang_for_(sub):
 
 def network_lopf(n, snapshots=None, solver_name="cbc",
          solver_logfile=None, extra_functionality=None, skip_objective=False,
-         skip_pre=False, extra_postprocessing=None, formulation="kirchhoff",
+         extra_postprocessing=None, formulation="kirchhoff",
          keep_references=False, keep_files=False,
          keep_shadowprices=['Bus', 'Line', 'Transformer', 'Link', 'GlobalConstraint'],
          solver_options=None, warmstart=False, store_basis=False,
-         solver_dir=None, ptdf_tolerance=0.):
```
Comment (fneum) [inline pypsa/linopf.py:825]: Yes, should be out.

File: pypsa/linopf.py (hunk 2 above)
Comment (FabianHofmann) [reply pypsa/linopf.py:902]: You're right, haven't thought about this. Let's keep it, even though I cannot imagine a case where you would skip it.

**Pull request merged**